tests/integration/Core/Framework/Adapter/Twig/fixtures/return-node/new-syntax/array-case.html.twig

```diff
@@ -0,0 +1,9 @@
+{%- macro arrayCaseNewSyntax(x) -%}
+    {%- if x == 1 -%}
+        1
```

Comment:
not sure if we should support it that way in app scripts: this was how twig macros worked before, for app scripts we add the return node to make it possible to pass the exact type object back to the script where you call it, see https://developer.shopware.com/docs/guides/plugins/apps/app-scripts/#including-scripts

therefore I would not support this syntax here, also this messes with how the macros on storefront twig rendering are executed, I wouldn't do that

Reply:
In short this "magic" behaviour (or hack) with returning the type instead of actually the result of the template was opt-in, you needed to use the return keyword to opt-in to that, we should keep it that way

Reply:
Sure, we can remove the tests, but currently, with the changes I did, both syntaxes are working.

Reply:
ah ok now i get it, it makes sense to have that to make sure it works as expected without the return node

can we add e.g. the array case for this as well to make sure without the return it is converted to string and our magic does not kick in?